packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -193,104 +202,106 @@ function translateWithDictionary(
   }
 
   const processedString =
-    typeof result === 'string' && pseudotranslate
+    isString(result) && pseudotranslate
       ? pseudotranslateString(result, {
           ...PSEUDOTRANSLATE_OPTIONS,
           toLocale:
             typeof pseudotranslate === 'boolean' ? undefined : pseudotranslate,
         })
       : result;
 
-  if (typeof processedString === 'string') {
-    return updateStringWithReplacements(processedString, {
-      ...replacements,
-      ...additionalReplacements,
-    });
-  } else {
+  if (!isString(processedString)) {
     return MISSING_TRANSLATION;
   }
+
+  return updateStringWithReplacements(
+    processedString,
+    {
+      ...replacements,
+      ...additionalReplacements,
+    },
+    {interpolate},
+  );
 }
 
+type UpdateStringWithReplacementsOptions = Pick<
+  TranslateOptions,
+  'interpolate'
+>;
+
 function updateStringWithReplacements(
   str: string,
   replacements?: ComplexReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): React.ReactElement<any>;
 function updateStringWithReplacements(
   str: string,
   replacements?: PrimitiveReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): string;
 function updateStringWithReplacements(
```

Comment:
I've simplified this function, so that there's only one path for every case now, instead of a special path for strings versus complex replacements.